### ISSUE TITLE
Update products for 23.7

### ISSUE THREAD

**Issue by sbernauer** [stackabletech/issues#375]
This ticket is used to collect the versions of upstream products we want to support in the next release.

In addition to the new and/or removed product versions we also want to update our base image to a new version at least once per release.

# Product versions

_Legend_:
* _Latest (Upstream)_: The latest upstream version within the product release line
* _Current (Stackable)_: The versions of the release line that we at Stackable currently support
* _New (Stackable)_: The versions of the release line that the next Stackable release will support

As of 2023-05-08

| Product   | Latest upstream version | Currently supported    | Added versions | Removed versions           | Deprecated versions      |
|-----------|-------------------------|------------------------|----------------|----------------------------|--------------------------|
| Airflow   | 2.6.0                   |                        | 2.6.0          |                            |                          |
| Airflow   | 2.5.3                   |                        |                |                            |                          |
| Airflow   | 2.4.3                   | 2.4.1                  |                |                            | 2.4.1                    |
| Airflow   | 2.3.3                   |                        |                |                            |                          |
| Airflow   | 2.2.5                   | 2.2.3, 2.2.4, 2.2.5    |                |                            | 2.2.3, 2.2.4, 2.2.5      |
| Druid     | 26.0.0                  |                        | 26.0.0         |                            |                          |
| Druid     | 24.0.2                  | 24.0.0                 |                |                            | 24.0.0                   |
| Druid     | 0.23.?                  | 0.23.0                 |                |                            | 0.23.0                   |
| HBase     | 2.5.4                   |                        | 2.5.4-hadoop3  |                            |                          |
| HBase     | 2.4.17                  | 2.4.6, 2.4.8, 2.4.9, 2.4.11, 2.4.12 |   |                            | 2.4.6, 2.4.8, 2.4.9, 2.4.11, 2.4.12 |
| HDFS      | 3.3.5                   | 3.3.1, 3.3.3, 3.3.4    | 3.3.5          |                            | 3.3.1, 3.3.3, 3.3.4      |
| HDFS      | 3.2.4                   | 3.2.2                  |                |                            | 3.2.2                    |
| Hive      | 3.1.3                   | 3.1.3                  |                |                            |                          |
| Hive      | 2.3.9                   | 2.3.9                  |                |                            | 2.3.9                    |
| Kafka     | 3.4.0                   |                        | 3.4.0          |                            |                          |
| Kafka     | 3.3.2                   | 3.3.1                  |                |                            | 3.3.1                    |
| Kafka     | 3.2.3                   | 3.2.0                  |                |                            | 3.2.0                    |
| Kafka     | 3.1.2                   | 3.1.0                  |                |                            | 3.1.0                    |
| Kafka     | 3.0.2                   |                        |                |                            |                          |
| Kafka     | 2.8.2                   | 2.8.1                  |                |                            | 2.8.1                    |
| Kafka     | 2.7.2                   | 2.7.1                  |                |                            | 2.7.1                    |
| NiFi      | 1.21.0                  |                        | 1.21.0         |                            |                          |
| NiFi      | 1.20.0                  |                        |                |                            |                          |
| NiFi      | 1.18.0                  | 1.18.0                 |                |                            | 1.18.0                   |
| NiFi      | 1.16.3                  | 1.16.0, 1.16.1, 1.16.2, 1.16.3 |        |                            | 1.16.0, 1.16.1, 1.16.2, 1.16.3 |
| NiFi      | 1.15.3                  | 1.15.0, 1.15.1, 1.15.2, 1.15.3 |        |                            | 1.15.0, 1.15.1, 1.15.2, 1.15.3 |
| OPA       | 0.51.0                  |                        | 0.51.0         |                            |                          |
| OPA       | 0.45.0                  | 0.45.0                 |                |                            | 0.45.0                   |
| OPA       | 0.41.0                  | 0.41.0                 |                |                            | 0.41.0                   |
| OPA       | 0.37.2                  | 0.37.2                 |                |                            | 0.37.2                   |
| OPA       | 0.28.0                  | 0.28.0                 |                |                            | 0.28.0                   |
| OPA       | 0.27.1                  | 0.27.1                 |                |                            | 0.27.1                   |
| Spark     | 3.4.0                   |                        | 3.4.0          |                            |                          |
| Spark     | 3.3.2                   | 3.3.0                  |                |                            | 3.3.0                    |
| Spark     | 3.2.4                   | 3.2.1                  |                |                            | 3.2.1                    |
| Superset  | 2.1.0                   |                        | 2.1.0          |                            |                          |
| Superset  | 2.0.1                   |                        |                |                            |                          |
| Superset  | 1.5.3                   | 1.5.1                  |                |                            | 1.5.1                    |
| Superset  | 1.4.2                   | 1.4.1                  |                |                            | 1.4.1                    |
| Superset  | 1.3.?                   | 1.3.2                  |                |                            | 1.3.2                    |
| Trino     | 414                     |                        | 414            |                            |                          |
| Trino     | 403                     | 403                    |                |                            | 403                      |
| Trino     | 396                     | 396                    |                |                            | 396                      |
| Trino     | 395                     | 395                    |                |                            | 395                      |
| Trino     | 387                     | 387                    |                |                            | 387                      |
| Trino     | 377                     | 377                    |                |                            | 377                      |
| ZooKeeper | 3.8.1                   | 3.8.0                  | 3.8.1          |                            | 3.8.0                    |
| ZooKeeper | 3.7.1                   | 3.7.0                  |                |                            | 3.7.0                    |
| ZooKeeper | 3.6.4                   | 3.6.3                  |                |                            | 3.6.3                    |
| ZooKeeper | 3.5.10                  | 3.5.8                  |                |                            | 3.5.8                    |

This will end up as the following supported versions in the next release:

| Product   | Supported                    | Supported (but deprecated)                                                         |
|-----------|------------------------------|------------------------------------------------------------------------------------|
| Airflow   | 2.6.0 (lts - 1 release)                  | 2.2.3, 2.2.4, 2.2.5, 2.4.1                                             |
| Druid     | 26.0.0 (lts - 1 release)                 | 0.23.0, 24.0.0                                                         |
| HBase     | 2.5.4-hadoop3 (lts - 1 release)          | 2.4.6, 2.4.8, 2.4.9, 2.4.11, 2.4.12                                    |
| HDFS      | 3.3.5 (lts - 1 release)                  | 3.2.2, 3.3.1, 3.3.3, 3.3.4                                             |
| Hive      | 3.1.3 (lts - 1 release)                  | 2.3.9                                                                  |
| Kafka     | 3.4.0 (lts - 1 release)                  | 2.7.1, 2.8.1, 3.1.0, 3.2.0, 3.3.1                                      |
| NiFi      | 1.21.0 (lts - 1 release)                 | 1.15.0, 1.15.1, 1.15.2, 1.15.3, 1.16.0, 1.16.1, 1.16.2, 1.16.3, 1.18.0 |
| OPA       | 0.51.0 (lts - 1 release)                 | 0.27.1, 0.28.0, 0.37..2, 0.41.0, 0.45.0                                |
| Spark     | 3.4.0 (lts - 1 release)                  | 3.2.1, 3.3.0                                                           |
| Superset  | 2.1.0 (lts - 1 release)                  | 1.3.2, 1.4.1, 1.5.1                                                    |
| Trino     | 414 (lts - 1 release)                    | 377, 387, 395, 396, 403                                                |
| ZooKeeper | 3.8.1 (lts - 1 release)                  | 3.5.8, 3.6.3, 3.7.0, 3.8.0                                             |

## Acceptance

We want to bump old and new product versions with the latest ubi image releases. This means bumping the stackable version and rerunning the build actions. This includes our [Java base image](https://github.com/stackabletech/docker-images/tree/main/java-base) which should be done before updating any Java based products.

### Base images

- [x] Java base image
- [x] Check if JMX Prometheus Exporter was updated: https://github.com/prometheus/jmx_exporter
- [x] https://github.com/stackabletech/airflow-operator/issues/281
- [x] https://github.com/stackabletech/druid-operator/issues/371
- [ ] https://github.com/stackabletech/hbase-operator/issues/352
- [x] Hadoop HDFS - not needed
- [x] Hive - not needed
- [x] https://github.com/stackabletech/kafka-operator/issues/590
- [x] https://github.com/stackabletech/nifi-operator/issues/403
- [x] https://github.com/stackabletech/opa-operator/issues/449
- [x] https://github.com/stackabletech/spark-k8s-operator/issues/242
- [x] https://github.com/stackabletech/superset-operator/issues/237
- [x] https://github.com/stackabletech/trino-operator/issues/422
- [x] https://github.com/stackabletech/zookeeper-operator/issues/686

### New versions

This is a tasklist pointing to issues in the product repositories:

- [ ] ....

**Comment by fhennig** [comment]
@sbernauer works for me as a template!

**Comment by lfrancke** [comment]
For now please *ONLY* add:
- NiFi 1.21.X
- Superset 2.X
- HBase 2.5.X

Leave all others out, including minor updates and deprecations.
I want to wait until after our LTS strategy discussion mid-May.
We will probably want to commit to have every single version supported for 2-3 SDP releases, for that we need to track when a version was first released. Nothing to do now, just so I don't forget.

Also, now that I've used the template I have to say that it's hard to read what has actually changed.
In other words: The old columns (added & removed) were useful, we don't have to do it now, but for next time I'd like them back I think.

**Comment by lfrancke** [comment]
I'm afraid my comment was ambiguous, sorry.

For each of those three above I only want one single latest version. I'll try to add the specific versions later. I should have done so from the start. Sorry.

**Comment by maltesander** [comment]
@lfrancke Superset (1.4.2, 1.5.3, 2.0.1, 2.1.0) and Nifi (1.20.0, 1.21.0) are merged already. Should i remove the minor / patch versions that are not latest from supported versions?

**Comment by lfrancke** [comment]
Phoenix is missing from the table.

**Comment by soenkeliebau** [comment]
Closing this even though HBase is not yet done. But since this ticket tracks the "regular" update process as part of the release cycle we decided to mark this as done and track the HBase update separately as this is a larger issue that may not even be possible yet.
